coingecko/src/endpoint/price.ts

```diff
@@ -21,62 +21,104 @@ const presetTickers: { [ticker: string]: string } = {
   GRT: 'the-graph',
 }
 
-const convertFromTicker = async (config: Config, ticker: string, coinId: string) => {
-  if (typeof coinId !== 'undefined') return coinId.toLowerCase()
-
-  // Correct common tickers that are misidentified
-  if (ticker in presetTickers) {
-    return presetTickers[ticker]
-  }
+const blacklist = ['leocoin', 'farmatrust', 'freetip', 'compound-coin', 'uni-coin', 'unicorn-token']
 
+export const getCoinList = async (config: Config) => {
   const url = '/coins/list'
-
   const options = {
     ...config.api,
     url,
   }
-
   const response = await Requester.request(options, customError)
-
-  const coin = response.data.find((x: any) => x.symbol.toLowerCase() === ticker.toLowerCase())
-
-  if (typeof coin === 'undefined') {
-    return undefined
-  }
-
-  return coin.id.toLowerCase()
+  return response.data
 }
 
-export const execute: ExecuteWithConfig<Config> = async (request, config) => {
-  const validator = new Validator(request, customParams)
-  if (validator.error) throw validator.error
-
-  const jobRunID = validator.validated.id
-  const symbol = validator.validated.data.base
-  const coinid = validator.validated.data.coinid
+export const getPriceData = async (
+  config: Config,
+  ids: string,
+  currency: string,
+  marketCap = false,
+) => {
   const url = '/simple/price'
-  const market = validator.validated.data.quote
-  const coin = await convertFromTicker(config, symbol, coinid)
-
   const params = {
-    ids: coin,
-    vs_currencies: market,
+    ids,
+    vs_currencies: currency.toLowerCase(),
+    include_market_cap: marketCap,
   }
-
   const options = {
     ...config.api,
     url,
     params,
   }
+  const response = await Requester.request(options)
+  return response.data
+}
 
-  const response = await Requester.request(options, customError)
-  const result = Requester.validateResultNumber(response.data, [
-    coin.toLowerCase(),
-    market.toLowerCase(),
-  ])
+export const getIdtoSymbol = (symbols: string[], coinList: any) => {
+  return Object.fromEntries(
+    symbols.map((symbol) => {
+      if (symbol in presetTickers) {
+        return [presetTickers[symbol], symbol]
+      }
+      const coin = coinList.find(
+        (d: any) =>
+          d.symbol.toLowerCase() === symbol.toLowerCase() &&
+          !blacklist.includes(d.id.toLowerCase()),
+      )
+      return [coin.id, symbol]
+    }),
+  )
+}
+
+export interface Prices {
+  [symbol: string]: number
+}
+
+const getPayload = (symbols: string[], prices: Prices, quote: string) => {
+  const payloadEntries = symbols.map((symbol) => {
+    const key = symbol.toUpperCase()
+    const val = {
+      quote: {
+        [quote.toUpperCase()]: {
+          price: prices[symbol],
+        },
+      },
+    }
+    return [key, val]
+  })
 
+  const payload: ResponsePayload = Object.fromEntries(payloadEntries)
+  return payload
```

Comment:
I think we should return batch payloads in as similar of a structure as possible to their input params.

So in this case of price if you put in:
```
{
base: ['btc', 'eth', 'uni'],
quote: 'btc'
}
```

It would return values with the same order
```
payload: [413414, 241444, 223]
```
Turning it into a map would happen in the composite adapter.

Reply:
I don't agree. Current Payload format is very rich and flexible. Is the one CMC uses and is currently used inside Token Allocations.